### PR TITLE
Don't store generic values as comments [TG-3652]

### PR DESCRIPTION
**Pull request by thk123**
This would cause them to get lost when symex_target_equation merges ireps. This way we preserve the generic specialisation information.

### CODE REVIEW

**Comment by thk123** [comment]
This needs a TG bump with some updated tests :white_check_mark: (just waiting on CI)

**Review by mgudemann — APPROVED** [review]

File: jbmc/regression/generics/type_erasure/TestClass.java
Comment (majakusber) [inline jbmc/regression/generics/type_erasure/TestClass.java]: Missing line

File: jbmc/regression/generics/type_erasure/TestClass.java
Comment (majakusber) [inline jbmc/regression/generics/type_erasure/TestClass.java]: I'm not sure I understand. With type erasure it should work, right? So it is our bug that it doesn't? If yes, should that be resolved before we can pull the generics out of comments?

**Review by majakusber — COMMENTED** [review]
Looks good but I'm not sure if the tests prove that we can really do this, see my question in code.

File: jbmc/regression/generics/type_erasure/TestClass.java
Comment (thk123) [reply jbmc/regression/generics/type_erasure/TestClass.java]: Comment wasn't clear (will update) - but this actually fails both with and without the change.

Comment (thk123) [reply jbmc/regression/generics/type_erasure/TestClass.java]: Well because the body of Object.equals for GOTO functions looks like

```
to_return = NONDET(_Bool);
return to_return
```

Will raise an issue as if two objects are actually the same it should work!

Comment (thk123) [reply jbmc/regression/generics/type_erasure/TestClass.java]: Which is just a property of not using the any models for java.lang.Object - will see if I can find out how regression tests are meant to use the core models. Thanks for making me dig a bit!

**Review by majakusber — APPROVED** [review]

**Comment by thk123** [comment]
Forgot to post here - TG CI has failed (some generic types are not printing correctly) and this has fallen below another item in my priority list so is currently stalled. I endeavour to get back to this PR this week.

**Comment by peterschrammel** [comment]
@thk123, please rebase to get rid of the AWS windows failure (maybe after merging #2346 to avoid a second rebase).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 496a3996a04114252e1aa6d13bd75ebd763b8018).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/76965274

**Comment by thk123** [comment]
Rebased to fix linting and clang format issues.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: e1398bc314782c296e945ed8c8aee6bae10d8aa6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/76976927

**Pull request merged**
